### PR TITLE
net-misc/aria2: bump to 1.35.0

### PR DESCRIPTION
**Pull request by crabbedhaloablution**
Bump EAPI while we're at it.

Closes: https://bugs.gentoo.org/704086
Package-Manager: Portage-3.0.12, Repoman-3.0.2
Signed-off-by: Peter Alfredsen <crabbedhaloablution@icloud.com>

### CODE REVIEW

**Review by mgorny — CHANGES_REQUESTED** [review]
Tests fail:

```
FAIL: aria2c
============================================================================
Testsuite summary for aria2 1.35.0
============================================================================
# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See test/test-suite.log
Please report to https://github.com/aria2/aria2/issues
============================================================================

```

**Comment by crabbedhaloablution** [comment]
> Tests fail:
> 
> [...]

I get a pass. 
```
PASS: aria2c
============================================================================
Testsuite summary for aria2 1.35.0
============================================================================
# TOTAL: 1
# PASS:  1
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
```
my test/test-suite.log isn't very verbose, is yours any better?
emerge --info =net-misc/aria2-1.35.0 would be helpful. Mine attached for reference:
```
saga /var/db/repos/alfsgard/net-misc/aria2 # emerge --info =net-misc/aria2-1.35.0
Portage 3.0.12 (python 3.8.7-candidate-1, default/linux/amd64/17.1/desktop/gnome/systemd, gcc-10.2.0, glibc-2.32-r5, 5.9.14 x86_64)
=================================================================
                         System Settings
=================================================================
System uname: Linux-5.9.14-x86_64-AMD_Ryzen_9_3900X_12-Core_Processor-with-glibc2.2.5
KiB Mem:   131829272 total,  81434104 free
KiB Swap:          0 total,         0 free
Timestamp of repository gentoo: Wed, 16 Dec 2020 19:45:01 +0000
Head commit of repository gentoo: e55c741babeea06f450934cfc91160a8dc4eadde
Head commit of repository alfsgard: 18c7d5da8e49cee9a1d2855fa11c6c2d92757560

Head commit of repository raspberrypi-mainline: 7ecd59c78d1bcf2adcaf6a2ba818f6d716c7f5ab

sh bash 5.1
ld GNU ld (Gentoo 2.35.1 p2) 2.35.1
distcc 3.3.3 x86_64-pc-linux-gnu [disabled]
app-shells/bash:          5.1::gentoo
dev-java/java-config:     2.3.1::gentoo
dev-lang/perl:            5.30.3-r1::gentoo
dev-lang/python:          2.7.18-r5::gentoo, 3.6.12::gentoo, 3.7.9::gentoo, 3.8.7_rc1::gentoo, 3.9.1::gentoo
dev-util/cmake:           3.19.1-r1::gentoo
sys-apps/baselayout:      2.7-r1::gentoo
sys-apps/sandbox:         2.20::gentoo
sys-devel/autoconf:       2.13-r1::gentoo, 2.69-r5::gentoo
sys-devel/automake:       1.16.3-r1::gentoo
sys-devel/binutils:       2.35.1-r1::gentoo
sys-devel/gcc:            10.2.0-r3::gentoo
sys-devel/gcc-config:     2.3.2-r1::gentoo
sys-devel/libtool:        2.4.6-r6::gentoo
sys-devel/make:           4.3::gentoo
sys-kernel/linux-headers: 5.10::gentoo (virtual/os-headers)
sys-libs/glibc:           2.32-r5::gentoo
Repositories:

gentoo
    location: /var/db/repos/gentoo
    sync-type: rsync
    sync-uri: rsync://rsync.de.gentoo.org/gentoo-portage
    priority: -1000
    sync-rsync-verify-jobs: 1
    sync-rsync-verify-max-age: 24
    sync-rsync-extra-opts: 
    sync-rsync-verify-metamanifest: yes

pf4public
    location: /var/lib/layman/pf4public
    sync-type: laymansync
    sync-uri: https://github.com/PF4Public/gentoo-overlay.git
    masters: gentoo
    priority: 50

steam-overlay
    location: /var/lib/layman/steam-overlay
    sync-type: laymansync
    sync-uri: https://github.com/anyc/steam-overlay.git
    masters: gentoo
    priority: 50

alfsgard
    location: /var/db/repos/alfsgard
    sync-type: git
    sync-uri: git@github.com:crabbedhaloablution/alfsgard.git
    masters: gentoo
    priority: 999

raspberrypi-mainline
    location: /var/db/repos/raspberrypi-mainline
    sync-type: git
    sync-uri: git@github.com:crabbedhaloablution/raspberrypi-mainline.git
    masters: gentoo
    priority: 999

cross-aarch64-unknown-linux-gnu
    location: /var/db/repos/cross-aarch64-unknown-linux-gnu
    masters: gentoo
    priority: 1000

saga
    location: /var/db/repos/saga
    masters: gentoo
    priority: 1000

ACCEPT_KEYWORDS="amd64 ~amd64"
ACCEPT_LICENSE="@FREE"
CBUILD="x86_64-pc-linux-gnu"
CFLAGS="-march=native -mtune=native -ggdb -O2 -pipe"
CHOST="x86_64-pc-linux-gnu"
CONFIG_PROTECT="/etc /usr/lib64/libreoffice/program/sofficerc /usr/share/gnupg/qualified.txt"
CONFIG_PROTECT_MASK="/etc/ca-certificates.conf /etc/dconf /etc/env.d /etc/fonts/fonts.conf /etc/gconf /etc/gentoo-release /etc/revdep-rebuild /etc/sandbox.d /etc/terminfo"
CXXFLAGS="-march=native -mtune=native -ggdb -O2 -pipe"
DISTDIR="/var/cache/distfiles"
EMERGE_DEFAULT_OPTS="--jobs 10 --load-average 18"
ENV_UNSET="CARGO_HOME DBUS_SESSION_BUS_ADDRESS DISPLAY GOBIN GOPATH PERL5LIB PERL5OPT PERLPREFIX PERL_CORE PERL_MB_OPT PERL_MM_OPT XAUTHORITY XDG_CACHE_HOME XDG_CONFIG_HOME XDG_DATA_HOME XDG_RUNTIME_DIR"
FCFLAGS="-march=native -mtune=native -ggdb -O2 -pipe"
FEATURES="assume-digests binpkg-docompress binpkg-dostrip binpkg-logs buildpkg config-protect-if-modified distlocks ebuild-locks fixlafiles installsources ipc-sandbox merge-sync multilib-strict network-sandbox news parallel-fetch pid-sandbox preserve-libs protect-owned qa-unresolved-soname-deps sandbox sfperms splitdebug strict unknown-features-warn unmerge-logs unmerge-orphans userfetch userpriv usersandbox usersync xattr"
FFLAGS="-march=native -mtune=native -ggdb -O2 -pipe"
GENTOO_MIRRORS="http://mirror.leaseweb.com/gentoo/  		http://ftp-stud.hs-esslingen.de/pub/Mirrors/gentoo/  		http://mirror.netcologne.de/gentoo/  		http://gentoo.mirrors.ovh.net/gentoo-distfiles/  		http://mirror.eu.oneandone.net/linux/distributions/gentoo/gentoo/  		http://linux.rz.ruhr-uni-bochum.de/download/gentoo-mirror/  		http://mirrors.soeasyto.com/distfiles.gentoo.org/  		http://ftp.halifax.rwth-aachen.de/gentoo/  		http://ftp.fau.de/gentoo"
LANG="en_GB.UTF-8"
LDFLAGS="-Wl,--as-needed"
MAKEOPTS="-j24"
PKGDIR="/var/cache/binpkgs"
PORTAGE_COMPRESS="zstd"
PORTAGE_COMPRESS_FLAGS="--quiet --rm -2 --threads=12"
PORTAGE_CONFIGROOT="/"
PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --omit-dir-times --compress --force --whole-file --delete --stats --human-readable --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages --exclude=/.git"
PORTAGE_TMPDIR="/var/tmp"
USE="X a52 aac acl acpi alsa amd64 berkdb bluetooth branding bzip2 cacert cairo caps cdda cdr cli colord crypt cups dbus dri dts dvd dvdr eds emboss encode evo exif faad fdk flac fortran frei0r fuse gdbm gif git gnome gnome-keyring gnome-online-accounts gnuefi gold google gpg gphoto2 gpm gstreamer gtk gui iconv icu idn introspection ios ipv6 jpeg lcms libdrm libglvnd libnotify libsecret libtirpc lm-sensors lzma mad matroska mercurial minizip mng mp3 mp4 mpeg multilib nautilus ncurses networkmanager nls nptl ntp nvme offensive ogg opengl openmp opus pam pango pcre pdf png policykit postproc ppds pulseaudio python qt5 readline sdl seccomp sndfile speex spell split-usr ssh ssl startup-notification svg sysprof systemd tcpd theora tiff tracker truetype twolame udev udisks uefi unicode upower usb user-session vaapi vdpau vorbis vpx vulkan webp widgets wxwidgets x264 x265 xattr xcb xml xv xvfb xvid zlib" ABI_X86="32 64" ADA_TARGET="gnat_2018" ALSA_CARDS="ali5451 als4000 atiixp atiixp-modem bt87x ca0106 cmipci emu10k1x ens1370 ens1371 es1938 es1968 fm801 hda-intel intel8x0 intel8x0m maestro3 trident usb-audio via82xx via82xx-modem ymfpci" APACHE2_MODULES="authn_core authz_core socache_shmcb unixd actions alias auth_basic authn_alias authn_anon authn_dbm authn_default authn_file authz_dbm authz_default authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate dir disk_cache env expires ext_filter file_cache filter headers include info log_config logio mem_cache mime mime_magic negotiation rewrite setenvif speling status unique_id userdir usertrack vhost_alias" CALLIGRA_FEATURES="karbon sheets words" COLLECTD_PLUGINS="df interface irq load memory rrdtool swap syslog" CPU_FLAGS_X86="aes avx avx2 f16c fma3 mmx mmxext pclmul popcnt sha sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3" ELIBC="glibc" GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock greis isync itrax mtk3301 nmea ntrip navcom oceanserver oldstyle oncore rtcm104v2 rtcm104v3 sirf skytraq superstar2 timing tsip tripmate tnt ublox ubx" INPUT_DEVICES="libinput" KERNEL="linux" LCD_DEVICES="bayrad cfontz cfontz633 glk hd44780 lb216 lcdm001 mtxorb ncurses text" LIBREOFFICE_EXTENSIONS="presenter-console presenter-minimizer" LUA_SINGLE_TARGET="lua5-1" LUA_TARGETS="lua5-1" OFFICE_IMPLEMENTATION="libreoffice" PHP_TARGETS="php7-2 php7-3 php7-4" POSTGRES_TARGETS="postgres10 postgres11" PYTHON_SINGLE_TARGET="python3_8 python2_7" PYTHON_TARGETS="python2_7 python3_8" RUBY_TARGETS="ruby27" USERLAND="GNU" VIDEO_CARDS="amdgpu radeonsi" XTABLES_ADDONS="quota2 psd pknock lscan length2 ipv4options ipset ipp2p iface geoip fuzzy condition tee tarpit sysrq steal rawnat logmark ipmark dhcpmac delude chaos account"
Unset:  CC, CPPFLAGS, CTARGET, CXX, INSTALL_MASK, LC_ALL, LINGUAS, PORTAGE_BINHOST, PORTAGE_BUNZIP2_COMMAND, PORTAGE_RSYNC_EXTRA_OPTS

=================================================================
                        Package Settings
=================================================================

net-misc/aria2-1.35.0::alfsgard was built with the following:
USE="gnutls libxml2 nettle nls ssh ssl -adns -bittorrent -jemalloc -libuv -metalink -scripts -sqlite -tcmalloc -test -xmlrpc" ABI_X86="(64)"
FEATURES="config-protect-if-modified sfperms binpkg-dostrip pid-sandbox ipc-sandbox xattr protect-owned unmerge-logs binpkg-logs usersandbox parallel-fetch network-sandbox preserve-libs userfetch unmerge-orphans splitdebug news merge-sync distlocks ebuild-locks userpriv qa-unresolved-soname-deps multilib-strict strict binpkg-docompress assume-digests installsources sandbox unknown-features-warn fixlafiles buildpkg usersync"
```

**Comment by crabbedhaloablution** [comment]
Ok, I can reproduce with all the use-flags, looks like a sandbox thing.
Let me throw cpu power at it for a bit and see what comes out.

**Comment by mgorny** [comment]
I think it's network sandbox, last I checked. I think disabling the two tests should fix it (but how?)

**Comment by crabbedhaloablution** [comment]
Fixed it. The sed is not pretty looking, could be reduced to a one-liner if you like that better, but I chose to robustify it so it is more likely to survive when upstream bumps their automake or adds another test on next release.

File: net-misc/aria2/aria2-1.35.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	if use scripts && ! use xmlrpc; then
+		ewarn "Please note that you may need to enable USE=xmlrpc to run the aria2rpc"
+		ewarn "and aria2mon scripts against the local aria2."
+	fi
+}
+
+src_prepare() {
+	default
+	sed -i -e "s|/tmp|${T}|" test/*.cc test/*.txt || die "sed failed"
+	# Fails with all-the-use-flags && FEATURES=network-sandbox
```
Comment (mgorny) [inline net-misc/aria2/aria2-1.35.0.ebuild:99]: I dare guess it's USE=bittorrent.

File: net-misc/aria2/aria2-1.35.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+	)
+	tcmalloc? ( dev-util/google-perftools )
+	xmlrpc? (
+		libxml2? ( >=dev-libs/libxml2-2.6.26:2= )
+		!libxml2? ( dev-libs/expat:0= ) )"
+
+DEPEND="${CDEPEND}
+	app-arch/xz-utils
+	virtual/pkgconfig
+	nls? ( sys-devel/gettext )
```
Comment (mgorny) [inline net-misc/aria2/aria2-1.35.0.ebuild:77]: These tree are BDEPEND in EAPI 7.

File: net-misc/aria2/aria2-1.35.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+#    5) !*? -> bundled md routines (we don't use them)
+# 3. There's also gmp usage for bittorrent with nettle.
+# 4. You can't really control features, just dependencies.
+# (we are skipping native Apple/Windows TLS support)
+#
+# We map this into:
+# ssl? -> openssl || (gnutls + (nettle || libgcrypt ))
+# !ssl? -> nettle || libgcrypt
+
+CDEPEND="sys-libs/zlib:0=
```
Comment (mgorny) [inline net-misc/aria2/aria2-1.35.0.ebuild:33]:
```suggestion
RDEPEND="sys-libs/zlib:0=
```

File: net-misc/aria2/aria2-1.35.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+			)
+		)
+		!nettle? ( >=dev-libs/libgcrypt-1.2.2:0= )
+	)
+	tcmalloc? ( dev-util/google-perftools )
+	xmlrpc? (
+		libxml2? ( >=dev-libs/libxml2-2.6.26:2= )
+		!libxml2? ( dev-libs/expat:0= ) )"
+
+DEPEND="${CDEPEND}
```
Comment (mgorny) [inline net-misc/aria2/aria2-1.35.0.ebuild:74]:
```suggestion
DEPEND=${RDEPEND}
```

File: net-misc/aria2/aria2-1.35.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+	app-arch/xz-utils
+	virtual/pkgconfig
+	nls? ( sys-devel/gettext )
+	test? ( >=dev-util/cppunit-1.12.0:0 )"
+RDEPEND="${CDEPEND}
```
Comment (mgorny) [inline net-misc/aria2/aria2-1.35.0.ebuild:79]:
```suggestion
RDEPEND+="
```

File: net-misc/aria2/aria2-1.35.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+	nls? ( virtual/libiconv virtual/libintl )
+	scripts? ( dev-lang/ruby )"
+
+# xmlrpc has no explicit switch, it's turned out by any XML library
+# so metalink implicitly forces it on
+REQUIRED_USE="?? ( jemalloc tcmalloc )
+	metalink? ( xmlrpc )"
+RESTRICT="!test? ( test )"
```
Comment (mgorny) [inline net-misc/aria2/aria2-1.35.0.ebuild:87]: Move these just below IUSE, please.

File: net-misc/aria2/aria2-1.35.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+	default
+	sed -i -e "s|/tmp|${T}|" test/*.cc test/*.txt || die "sed failed"
+	# Fails with all-the-use-flags && FEATURES=network-sandbox
+	sed -i -E \
+		-e '/^@ENABLE_BITTORRENT_TRUE@	LpdMessageDispatcherTest/d' \
+		-e '/^@ENABLE_BITTORRENT_TRUE@	LpdMessageReceiverTest/d' \
+		-e 's|LpdMessageDispatcherTest.cc||' \
+		-e 's|LpdMessageReceiverTest.cc||' \
+		-e '/LpdMessageDispatcherTest.Po/d' \
+		-e '/LpdMessageReceiverTest.Po/d' \
```
Comment (mgorny) [inline net-misc/aria2/aria2-1.35.0.ebuild:106]: Do you actually need this? I don't think this affects the targets we're using.

File: net-misc/aria2/aria2-1.35.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+		test/Makefile.in||die
```
Comment (mgorny) [inline net-misc/aria2/aria2-1.35.0.ebuild:107]:
```suggestion
		test/Makefile.in || die
```

File: net-misc/aria2/aria2-1.35.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+	# - always enable gzip/http compression since zlib should always be available anyway
+	# - always enable epoll since we can assume kernel 2.6.x
+	# - other options for threads: solaris, pth, win32
+	econf "${myconf[@]}"
+}
+
+src_install() {
+	default
+	rm -rf "${D}"/usr/share/doc/aria2 \
+		"${D}"/usr/share/doc/${PF}/README{,.html}
```
Comment (mgorny) [inline net-misc/aria2/aria2-1.35.0.ebuild:179]:
```suggestion
	rm -r "${ED}"/usr/share/doc/aria2 \
		"${ED}"/usr/share/doc/${PF}/README{,.html} || die
```

**Review by mgorny — CHANGES_REQUESTED** [review]

**Comment by crabbedhaloablution** [comment]
Re-pushed, all requests integrated.

File: net-misc/aria2/aria2-1.35.0.ebuild [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+		ewarn "Please note that you may need to enable USE=xmlrpc to run the aria2rpc"
+		ewarn "and aria2mon scripts against the local aria2."
+	fi
+}
+
+src_prepare() {
+	default
+	sed -i -e "s|/tmp|${T}|" test/*.cc test/*.txt || die "sed failed"
+	# Fails with USE=bittorrent && FEATURES=network-sandbox
+	sed -i -E \
```
Comment (mgorny) [inline net-misc/aria2/aria2-1.35.0.ebuild:98]: I'm sorry for missing this previously but I don't think you're actually using extended regexp syntax here.

**Comment by mgorny** [comment]
Overall good work, thanks! If you want me to fix that last nit, please just confirm that I'm correct and I'll do it while merging.

File: net-misc/aria2/aria2-1.35.0.ebuild (hunk 10 above)
Comment (crabbedhaloablution) [reply net-misc/aria2/aria2-1.35.0.ebuild:98]: You're correct. I was experimenting with a one-liner to start with, but felt better with a readable solution.

**Review by mgorny — APPROVED** [review]
Thanks. I've had to kill that `/usr/share/doc/${PN}` rm because no such directory existed, and then I merged it.

**Pull request closed without merging**